src/components/SettlementButton.tsx

```diff
@@ -1,143 +1,106 @@
-import PropTypes from 'prop-types';
-import React, {useEffect, useMemo} from 'react';
-import {withOnyx} from 'react-native-onyx';
-import _ from 'underscore';
+import _ from 'lodash';
+import React, {MutableRefObject, useEffect, useMemo} from 'react';
+import {StyleProp, ViewStyle} from 'react-native';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
 import useLocalize from '@hooks/useLocalize';
 import useNetwork from '@hooks/useNetwork';
-import compose from '@libs/compose';
 import * as ReportUtils from '@libs/ReportUtils';
-import iouReportPropTypes from '@pages/iouReportPropTypes';
 import * as BankAccounts from '@userActions/BankAccounts';
 import * as IOU from '@userActions/IOU';
 import * as PaymentMethods from '@userActions/PaymentMethods';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import {ButtonSizeValue} from '@src/styles/utils/types';
+import type {Report} from '@src/types/onyx';
+import {EmptyObject} from '@src/types/utils/EmptyObject';
 import ButtonWithDropdownMenu from './ButtonWithDropdownMenu';
 import * as Expensicons from './Icon/Expensicons';
 import KYCWall from './KYCWall';
-import withNavigation from './withNavigation';
 
-const propTypes = {
-    /** Callback to execute when this button is pressed. Receives a single payment type argument. */
-    onPress: PropTypes.func.isRequired,
+type AnchorAlignment = {
+    horizontal: 'left' | 'right' | 'center';
+    vertical: 'top' | 'center' | 'bottom';
+};
 
-    /** Call the onPress function on main button when Enter key is pressed */
-    pressOnEnter: PropTypes.bool,
+type TriggerKYCFlow = (event: Event, iouPaymentType: string) => void;
 
-    /** Settlement currency type */
-    currency: PropTypes.string,
+type SettlementButtonOnyxProps = {
+    /** The last payment method used per policy */
+    nvpLastPaymentMethod?: Record<string, string>;
+};
 
+type SettlementButtonProps = SettlementButtonOnyxProps & {
+    /** Callback to execute when this button is pressed. Receives a single payment type argument. */
+    onPress: (paymentType: string) => void;
+    /** The route to redirect if user does not have a payment method setup */
+    enablePaymentsRoute: string;
```

Comment:
I think this shouldn't be just any string but we should limit this to only few values. I found those used for this prop
`ROUTES.ENABLE_PAYMENTS`, `ROUTES.IOU_SEND_ENABLE_PAYMENTS`, `ROUTES.SETTINGS_ENABLE_PAYMENTS`